prog/vm/nexus.rb

```diff
@@ -399,25 +411,61 @@ def write_params_json
 
     vm.update(display_state: "deleting")
 
+    clean_up_host
+    give_back_resources
+
+    hop_wait_lb_expiry if vm.load_balancer
+
+    final_clean_up
+
+    pop "vm deleted"
+  end
+
+  label def recreate
+    decr_recreate
+    clean_up_host(force_network_cleanup: true, omit_failure: true)
+    if vm.load_balancer
+      vm.load_balancer.evacuate_vm(vm)
+      vm.load_balancer.remove_vm(vm)
+    end
+
+    give_back_resources
+    vm.assigned_vm_address&.destroy
+    vm.vm_storage_volumes.each(&:destroy)
+    DB.transaction do
+      vm.nics.map do |nic|
+        ps = nic.private_subnet
+        nic.update(vm_id: nil)
+        nic.incr_destroy
+        new_nic = Prog::Vnet::NicNexus.assemble(ps.id, name: "#{vm.name}-nic").subject
```

Comment:
Are there any limits on naming?  If Vm and Nic names have the same limits, there would likely be problems if the Vm name was near the limit.

Reply:
Definitely there are limits, though the vm name was chosen to be as short as possible while not engendering excessive collisions. Different systems in Linux have different limits as well (e.g. interface names, user names), so we kind of needed the minimum of all of them, plus some space for derivatives such as this one.

Reply:
Also, we don't use the nic name directly in the server, rather make use of the first 7 chars of the ubid. So, this name is more for us or the customer when we expose them.